0004-ledger-plugin-interface/0004-ledger-plugin-interface.md

```diff
@@ -336,7 +338,7 @@ If the `data` is too large, the ledger plugin MUST throw a `MaximumDataSizeExcee
 
 An optional bytestring containing details the host needs to persist with the transfer in order to be able to react to transfer events like condition fulfillment later.
 
-Ledger plugins MAY attach the `noteToSelf` to the transfer and let the ledger store it. Otherwise it MAY use the [`store`](#TODO) in order to persist this field. Regardless of the implementation, the ledger plugin MUST ensure that all instances of the transfer carry the same `noteToSelf`, even across different machines.
+Ledger plugins MAY attach the `noteToSelf` to the transfer and let the ledger store it. Otherwise it MAY use the [`store`](#store) in order to persist this field. Regardless of the implementation, the ledger plugin MUST ensure that all instances of the transfer carry the same `noteToSelf`, even across different machines.
 
```

Comment:
FYI, this is a well-used concept in ISO8583 (card) messaging.
DE 59 in the Postilion ISO8583 implementation was called "echo data". Whatever you put in there in a request was returned in the same place in a response. I've also seen it used in HSM comms protocols that have a duplex streaming protocol so that you can easily match a request and response pair.